fix(language_server): handle invalid root URI gracefully

381181295 opened the pull request:
## Summary

- Server panicked on init when `rootUri` is invalid (e.g., `file://` with no path)
- `to_file_path().unwrap()` failed in `server_formatter.rs` and `server_linter.rs`
- Changed `ToolBuilder::build_boxed` to return `Option<Box<dyn Tool>>`
- Worker now uses `filter_map()` to skip tools that fail to build
- Server responds with capabilities instead of crashing

## Repro (before fix)

```bash
echo 'Content-Length: 95

{"jsonrpc":"2.0","id":0,"method":"initialize","params":{"capabilities":{},"rootUri":"file://"}}' | oxc_language_server
# Panics: called Option::unwrap() on a None value
```

## After fix

Server returns capabilities JSON and continues running.

381181295 committed:
fix(language_server): handle invalid root URI gracefully

When LSP clients send an invalid rootUri (e.g., 'file://' with no path),
to_file_path() returns None. Previously this caused a panic via unwrap().

Now build_boxed() returns Option<Box<dyn Tool>>, allowing worker to skip
tools that fail to initialize. Server responds with capabilities instead
of crashing.

Sysix reviewed (commented):
Can you explain a bit more why this is needed? 
I would never expect an invalid file URI for workspace/root URI's:) 

@camc314  I do not think we will support non-file URIs for a tool base.
If we are 100% sure, we should probably add a check in the `initialize` request and return an `Err` instead of panicking.

camc314 reviewed (approved):
Even if we don't know what case causes this, it seems reasonable to handle by retuning an err (over unwrapping)

camc314 committed:
Merge branch 'main' into fix/language-server-invalid-root-uri

381181295 committed:
fix(language_server): use file path comparison in is_responsible_for_uri

String prefix matching incorrectly matched file:///path/test_workspace_second
against root file:///path/test_workspace. Path::starts_with respects
directory boundaries.

381181295 committed:
test(language_server): add regression test for root_second prefix collision

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

Sysix commented:
Because `is_responsible_for_uri`  uses again for `to_file_path` on the workspace worker root URI, no workspace worker is responsible for a URI ⇒ No tasks are done.

>  this was causing root_second to incorrectly match root.
I guess we need to sort the workspace workers before setting them in `Backend`. Or add a trailing slash at the URI, so it does not match similar names, like `root` and `root_xyz`

After https://github.com/oxc-project/oxc/pull/17388 I am not sure, if this PR is needed anymore. 
@camc314 your thoughts? Fix sorting/trailing or closing this PR :)?
